improve documentation (#1046)
Proteobench/ProteoBench: docs/general-information/6-faq.md

# Frequently asked questions

> ## What is ProteoBench?

Reviewer: For this section "What is ProteoBench", and the following "who is PB intended for?", I would merge it with the text already here: https://proteobench.readthedocs.io/en/stable/general-information/0-about/
We put a lot of thoughts into it. 

Any idea how to best merge the two?

The goal of ProteoBench is not to prescribe a single best workflow, but to make workflow evaluation more transparent, reproducible, and comparable across tools, laboratories, and software versions. 

## Who is ProteoBench intended for?

ProteoBench is intended for several user groups:

- Proteomics researchers who want to compare their local workflows against public reference results.
- Proteomics researchers who want to find the best workflow for their use case
- Core facilities that want to evaluate, document, or monitor routine analysis pipelines.
- Software developers who want to test new algorithms or software versions on standardized datasets.
- Reviewers, editors, and readers who want transparent evidence for workflow performance claims.
- Method developers who want to contribute new benchmark modules for emerging proteomics applications.

Although ProteoBench is useful for software developers, it is also designed for experimental laboratories and core facilities that use established open-source or commercial software.

> ## Do I need to use a predefined dataset?

Reviewer: ```suggestion
## Do I need to run my workflow on a predefined data set?
```

Yes. To compare results meaningfully within a ProteoBench module, all submitted workflows must be evaluated on the benchmark dataset defined for that module. This ensures that differences between submitted results reflect differences in data analysis workflows rather than differences in sample composition, instrument configuration, acquisition design, or data quality.

Users who want to benchmark a different experimental design can contribute or request a new module. ProteoBench is modular by design, so new datasets, metrics, and use cases can be added as the community’s needs evolve.

> ## Do I need to use the exact same search parameters as other users?

Reviewer: ```suggestion
## Do I need to use the exact same workflow parameters as other users?
```

> No. ProteoBench does not require all users to run identical search parameters. In many modules, suggested parameters are provided to help users get started and to support comparable baseline analyses, but users may submit results generated with their own parameter choices.

Reviewer: ```suggestion
No. ProteoBench does not require all users to run identical workflow parameters. In many modules, suggested parameters are provided to help users get started and to support comparable baseline analyses, but users may submit results generated with their own parameter choices.
```

> Parameter information is collected with submissions whenever possible so that users can interpret differences in performance in the context of software versions, workflow settings, search databases, quantification options, and other relevant configuration details. If you 'limit test' a specific parameter, we recommend documenting that in the comment field.

Reviewer: ```suggestion
The parameter used for all the public :ref:`benchmark runs<benchmark-run>` is collected upon submission and fully available to download so that users can interpret differences in performance in the context of software versions, workflow settings, search databases, quantification options, and other relevant configuration details. If you 'limit test' a specific parameter, we recommend documenting that in the comment field.
```

## Do I need to use a specific FASTA database?

For modules where database choice affects the benchmark outcome, the recommended or required FASTA database is specified in the module documentation. Using the same database helps ensure that submitted results are comparable.

## Can I benchmark commercial software?

> Yes. ProteoBench can include results from commercial software when the software output can be parsed or converted into one of the supported submission formats. Commercial software results serve as reference points that can be compared with workflows generated using accessible open-source tools.

Reviewer 2: I think "Commercial software results serve as reference points that can be compared with workflows generated using accessible open-source tools." will come across quite confusing

If anything it's the reverse, no?

Author: the phrasing is a bit bad yeah

Author: removed the sentence

## What should I do if my software is not directly supported?

If a tool-specific parser is not yet available, users can submit results using the custom tabular format described in the module documentation. This format defines the required columns for metric calculation and enables benchmarking of tools that are not yet natively supported.

Users and developers are encouraged to contact the ProteoBench community if they want a new parser to be added for a specific software package.

## Why are submissions reviewed before becoming public?

Manual review is used as a quality-control step to keep public benchmark results interpretable and reliable. The review process helps detect incomplete submissions, incorrect file types, incompatible module choices, missing metadata, or results that cannot be meaningfully compared with existing entries.

This step is not intended to restrict participation. Rather, it protects the benchmark from accidental or automated uploads that could make the public comparison plots misleading.

> ## Does ProteoBench run my workflow automatically?

Reviewer: ❤️

> This step is not intended to restrict participation. Rather, it protects the benchmark from accidental or automated uploads that could make the public comparison plots misleading.

Reviewer 2: Do we discourage automated submissions? 
As long as the data ends up on server, it's fine no? (how one would automize submissions in a way that data would end up on server, I would be interested to know, but if they can figure it out I have no problem with automatic submissions)

Author: if someone automatically uploads 500 sage runs, what do we do? manually check them all?

Reviewer 2: I dont see why they are any more useless than if 500 people each submitted 1 point, so I guess we'll have to, yes

Author: then we need reference points, otherwise the main plot will not be useful anymore

Reviewer 2: Agreed, but that's future music
I just don't believe we should explicitely state that you cannot figure out a way to automatically submit points

Reviewer: I would remove this paragraph (line 52). I don't think that this is currently a risk. And I would favour simplicity in the answers.

> At present, ProteoBench primarily evaluates submitted workflow outputs rather than executing every workflow centrally. Users usually download the benchmark data, process it with their local software or pipeline, and upload the relevant output files.

Reviewer: ```suggestion
ProteoBench evaluates submitted workflow outputs rather than executing every workflow centrally. Users download the benchmark data provided for their module of interest, process it with their local software or pipeline, and upload the relevant output files.
```

> This design allows ProteoBench to support a wide range of tools, including commercial software, locally configured pipelines, and workflows under active development. Centralized execution of all supported tools would be difficult to maintain because software licenses, operating-system requirements, hardware needs, and workflow customization differ substantially across proteomics software ecosystems.

Reviewer: ```suggestion
This design allows ProteoBench to support a wide range of tools, including commercial software, locally configured pipelines, and workflows under active development. 
```

Reviewer: I would remove the justification because we do not have to justify ourselves in the FAQ (I think), and because this justification is not the only reason. 
ProteoBench was designed (originally) by non developers. And it always aimed at making people work together, facilitating comparison. Its purpose was never to run the pipelines.

> ## How do I find the raw data for a benchmark module?

Reviewer: ```suggestion
## How do I find the input data associated with a benchmark module?
```

> Each module should provide a clear route to the required benchmark files, including raw MS files, search databases when applicable, example outputs, and documentation. The raw data are typically linked from the module documentation or from public repositories such as ProteomeXchange, depending on the dataset.

Reviewer: ```suggestion
Each module provides a clear route to the required benchmark files, including raw MS files, search databases when applicable, example outputs, and documentation. The raw data are typically linked from the module documentation or from public repositories such as ProteomeXchange, depending on the dataset.
If you do not find the information that you need to run your workflow, don't hesitate to contact us.
```

> ## What does “depth” mean in ProteoBench?

Reviewer: Since we plan to use "sensitivity" in the paper, shouldn't we also use it here? And remove this question?

In ProteoBench, “depth” refers to the number of identifications or quantified features reported by a workflow within a given benchmark module. The exact unit depends on the module and may refer, for example, to precursor ions, peptides, peptide groups, proteins, or other analyte classes.

Because “depth” can mean different things in different proteomics contexts, users should consult the metric definitions in each module before comparing results.

## What is QuantError?

QuantError is a module-specific metric used to assess quantitative accuracy when the expected abundance ratios are known. It compares observed quantitative values against expected values and summarizes how far a workflow deviates from the known experimental design.

> QuantError should be interpreted together with identification depth and other module-specific metrics. A workflow that reports many quantified features is not necessarily optimal if the quantitative error is high.

Reviewer: ```suggestion
QuantError should be interpreted together with the other module-specific metrics. A workflow that reports many quantified features is not necessarily optimal if the quantitative error is high.
```

## Does ProteoBench perform normalization or missing-value imputation?

> ProteoBench generally evaluates the outputs provided by the submitted workflow. It does not aim to impose a uniform post-processing workflow across all tools, because normalization, missing-value handling, transfer steps, protein inference, and quantification strategies are often integral parts of the software being benchmarked.

Reviewer: ```suggestion
ProteoBench does not apply post-processing steps that may affect the benchmark metrics. We consider normalization, missing-value handling, transfer steps, protein inference, and quantification strategies as integral parts of benchmarked workflows. They should be reported in the parameter files submitted alongside the workflow results upon public submission.
```

> Where relevant, module documentation should specify which processing steps are performed by ProteoBench itself and which are expected to have been performed by the submitted workflow.

Reviewer: ```suggestion
Where relevant, module documentation specifies which processing steps are performed by ProteoBench itself and which are expected to have been performed by the submitted workflow.
```

> ## How are differences between software tools interpreted?

Reviewer: ```suggestion
## How can I interpret differences between :ref:`benchmark runs<benchmark-run>`?
```

Reviewer: I think that we could put the link to the "do and don't" paragraph in the "about proteobench" here.

ProteoBench provides standardized metrics and visualizations that make differences between workflows visible. These differences may arise from many factors, including feature detection, spectral library generation, identification scoring, false discovery rate control, match-transfer strategies, normalization, missing-value handling, quantification algorithms, and software-specific defaults.

ProteoBench results should therefore be interpreted as workflow-level comparisons, not only as comparisons of individual algorithms. Where possible, users should inspect both the software version and the reported parameters before drawing conclusions.

## Does ProteoBench validate false discovery rates independently?

False discovery rate control is a central concern in proteomics benchmarking. ProteoBench modules can incorporate FDR-related analyses when appropriate datasets and metrics are available. Independent validation strategies, including entrapment-based approaches, are especially relevant for workflows where reported identifications and quantification accuracy depend strongly on tool-specific error-control procedures.

> ## How should I interpret rankings or leaderboard-style plots?

Reviewer: Do we have those? I don't think so... I would remove this paragraph

Author: I think it is just ill named, otherwise imo its an important point to make that comparisons should be seen in context and not absolute

Reviewer: hum... But since we are careful not to rank the benchmarks. So if you want to keep it we can change the title of the section to "How should I interpret ProteoBench main plots"?
And then the answer goes down to what is currently in the [goals and no goals section of the documentation](https://proteobench.readthedocs.io/en/stable/general-information/0-about/#goals-and-non-goals)

> ProteoBench comparison plots should not be interpreted as universal rankings of software tools. A tool that performs well in one module, dataset, or metric may not be optimal for another biological question, instrument type, acquisition method, or analysis goal.

Reviewer: I would change to:

ProteoBench comparison plots should not be interpreted as universal rankings of software tools. A tool that performs well in one module, with one dataset or metric, may not be optimal for another biological question, instrument type, acquisition method, or analysis goal.

Users should consider:

- The benchmark module and dataset.
- The software version.
- The submitted parameters.
- The relevant metric definitions.

> - The trade-off between identification depth and quantitative accuracy.

Reviewer: this one is very specific to the current modules, isn't it?
Maybe we could instead mention "hidden" factors that would not be the focus of a given module (such as false discovery in the context of the quantification module)? Or maybe we should advise users to potentially explore several modules that would provide complementary information? 
What do you think?

Author: I made the last point more general, I think the other points apply to all modules so far. For the hidden factors, I added the FDR as an example to the parameter point

- Whether the workflow resembles their own intended use case.

ProteoBench is best used as a transparent decision-support resource rather than as a single definitive ranking system.

> ## How can I follow updates or community discussions?

Reviewer: I would add the resources links in this paragraph

Reviewer: I guess also mention that if people have a GitHub account they can already register to discussion updates.

ProteoBench development and discussion are community-oriented. Users can follow the relevant repository, documentation pages, discussion forum, or community channels to receive updates about new modules, software parsers, benchmark datasets, documentation improvements, and public releases.

## How can I contribute to ProteoBench?

> Users can contribute by submitting benchmark results, reporting issues, improving documentation, proposing new modules, contributing parsers for additional software tools, sharing tutorial material, or participating in community discussions.

Reviewer: ```suggestion
Users can contribute by submitting their workflow outputs, reporting issues, improving documentation, proposing new modules, contributing parsers for additional software tools, sharing tutorial material, or participating in community discussions.
```

## Does ProteoBench validate false discovery rates independently?

False discovery rate control is a central concern in proteomics benchmarking. ProteoBench modules can incorporate FDR-related analyses when appropriate datasets and metrics are available. Independent validation strategies, including entrapment-based approaches, are especially relevant for workflows where reported identifications and quantification accuracy depend strongly on tool-specific error-control procedures.

## How should I interpret the main plots in each module?

ProteoBench comparison plots should not be interpreted as universal rankings of software tools. A tool that performs well in one module, dataset, or metric may not be optimal for another biological question, instrument type, acquisition method, or analysis goal.

Users should consider:

- The benchmark module and dataset.
- The software version.
- The submitted parameters.
- The relevant metric definitions.
- The trade-off between identification depth and quantitative accuracy.
- Whether the workflow resembles their own intended use case.

> ProteoBench is best used as a transparent decision-support resource rather than as a single definitive ranking system.

Reviewer: This really comes down to the don't:

DOES NOT point to a single best one-fits-all data analysis workflow
SHOULD NOT be used as evidence for generalized statements about a workflow’s performance
SHOULD NOT be used by developers as single performance measure of their workflow

These could be repeated here?
We could say (instead of line 106):
"ProteoBench does not point to a single best one-fits-all data analysis workflow; should not be used as evidence for generalized statements about a workflow’s performance, and should not be used by developers as single performance measure of their workflow."

I insist on these because we actually put a lot of thinking in their phrasing.

# Frequently asked questions

## What is ProteoBench?

> ProteoBench is an open and collaborative platform for community-curated benchmarks for proteomics data analysis pipelines. Its goal is to enable continuous, easy, and controlled comparison of proteomics data analysis workflows.

Reviewer: ```suggestion
ProteoBench is an open and collaborative platform for community-curated benchmarks that enables continuous, easy, and controlled comparison of proteomics data analysis workflows.
```

Reviewer: we agreed early on that we would not use "pipeline" but "workflow"

> ProteoBench provides a centralized web platform where developers and end-users can compare proteomics data analysis pipelines using standardized benchmark datasets, transparent evaluation metrics, and public comparison plots. This community-curated effort supports the controlled evaluation of tools developed or used by participants against other state-of-the-art pipelines for specific proteomics applications. The ability to add new benchmark points and modules allows ProteoBench to grow with the proteomics field.

Reviewer: ```suggestion
It provides a centralized web platform where developers and end-users can compare proteomics data analysis {ref}`workflows<workflow>` using standardized benchmark datasets, transparent evaluation metrics, and public comparison plots. This community-curated effort supports the controlled evaluation of tools developed or used by participants against other state-of-the-art workflows for specific proteomics applications. The ability to add new benchmark runs and modules allows ProteoBench to grow with the proteomics field. (See our goals and non-goals [here](../general-information/0-about.html#goals-and-non-goals))
```

The goal of ProteoBench is not to select or prescribe a single best one-size-fits-all data analysis workflow. Instead, ProteoBench aims to make workflow evaluation more transparent, reproducible, and comparable across tools, laboratories, software versions, and use cases.

> ## Who is ProteoBench intended for?

Reviewer: ```suggestion
## Who is ProteoBench intended for?

```

ProteoBench is intended for several user groups involved in the development, evaluation, or application of proteomics data analysis workflows.

> For **end-users**, ProteoBench helps identify suitable workflows for specific analytical needs by allowing comparison with public reference results and state-of-the-art pipelines. This includes proteomics researchers, experimental laboratories, and core facilities that want to evaluate, document, or monitor routine analysis pipelines.

Reviewer: ```suggestion
For **end-users**, ProteoBench helps identify suitable workflows for specific analytical needs by allowing comparison with public reference results and state-of-the-art pipelines. This includes proteomics researchers, experimental laboratories, and core facilities that want to evaluate, document, or monitor routine analysis workflows.
```

## Do I need to run my workflow on a predefined data set?

Yes. To compare results meaningfully within a ProteoBench module, all submitted workflows must be evaluated on the benchmark dataset defined for that module. This ensures that differences between submitted results reflect differences in data analysis workflows rather than differences in sample composition, instrument configuration, acquisition design, or data quality.

Users who want to benchmark a different experimental design can contribute or request a new module. ProteoBench is modular by design, so new datasets, metrics, and use cases can be added as the community’s needs evolve.

## Do I need to use the exact same workflow parameters as other users?

No. ProteoBench does not require all users to run identical workflow parameters. In many modules, suggested parameters are provided to help users get started and to support comparable baseline analyses, but users may submit results generated with their own parameter choices.

> The parameter used for all the public :ref:`benchmark runs<benchmark-run>` is collected upon submission and fully available to download so that users can interpret differences in performance in the context of software versions, workflow settings, search databases, quantification options, and other relevant configuration details. If you 'limit test' a specific parameter, we recommend documenting that in the comment field.

Reviewer: ```suggestion
The parameters used for all the public {ref}`benchmark runs<benchmark-run>` is collected upon submission and fully available to download so that users can interpret differences in performance in the context of software versions, workflow settings, search databases, quantification options, and other relevant configuration details. If you 'limit test' a specific parameter, we recommend documenting that in the comment field when you submit your benchmark run for public release.
```

Reviewer: I fix my own mistakes, sorry.

## Do I need to use a specific FASTA database?

> For modules where database choice affects the benchmark outcome, the recommended or required FASTA database is specified in the module documentation. Using the same database helps ensure that submitted results are comparable.

Reviewer: ```suggestion
For modules where database choice affects the benchmark outcome, the recommended or required FASTA database is specified in the module documentation. Using the same database ensures that submitted results are comparable.
```

## Can I benchmark commercial software?

> Yes. ProteoBench can include results from commercial software when the software output can be parsed or converted into one of the supported submission formats. 

Reviewer: ```suggestion
Yes. ProteoBench is designed to include results from commercial software when the software outputs all the workflow parameters/metadata in a format that can be parsed or converted into one of the supported submission formats. 
```

Users and developers are encouraged to contact the ProteoBench community if they want a new parser to be added for a specific software package.

## Why are submissions reviewed before becoming public?

Manual review is used as a quality-control step to keep public benchmark results interpretable and reliable. The review process helps detect incomplete submissions, incorrect file types, incompatible module choices, missing metadata, or results that cannot be meaningfully compared with existing entries.

## Does ProteoBench run my workflow automatically?

ProteoBench evaluates submitted workflow outputs rather than executing every workflow centrally. Users download the benchmark data provided for their module of interest, process it with their local software or pipeline, and upload the relevant output files.

This design allows ProteoBench to support a wide range of tools, including commercial software, locally configured pipelines, and workflows under active development. 

## How do I find the input data associated with a benchmark module?

> Each module provides a clear route to the required benchmark files, including raw MS files, search databases when applicable, example outputs, and documentation. The raw data are typically linked from the module documentation or from public repositories such as ProteomeXchange, depending on the dataset.

Reviewer: ```suggestion
Each module provides a clear route to the required benchmark files, including raw MS files, search databases when applicable, example outputs, and documentation. The raw data are typically linked from the module documentation or from public repositories such as [ProteomeXchange](https://www.proteomexchange.org/), depending on the dataset.
```

If you do not find the information that you need to run your workflow, don't hesitate to contact us.

## What is QuantError?

QuantError is a module-specific metric used to assess quantitative accuracy when the expected abundance ratios are known. It compares observed quantitative values against expected values and summarizes how far a workflow deviates from the known experimental design.

QuantError should be interpreted together with the other module-specific metrics. A workflow that reports many quantified features is not necessarily optimal if the quantitative error is high.

## Does ProteoBench perform normalization or missing-value imputation?

ProteoBench does not apply post-processing steps that may affect the benchmark metrics. We consider normalization, missing-value handling, transfer steps, protein inference, and quantification strategies as integral parts of benchmarked workflows. They should be reported in the parameter files submitted alongside the workflow results upon public submission.

Where relevant, module documentation specifies which processing steps are performed by ProteoBench itself and which are expected to have been performed by the submitted workflow.

> ## How can I interpret differences between :ref:`benchmark runs<benchmark-run>`?

Reviewer: ```suggestion
## How can I interpret differences between {ref}`benchmark runs<benchmark-run>`?
```

## How should I interpret the main plots in each module?

ProteoBench comparison plots should not be interpreted as universal rankings of software tools. A tool that performs well in one module, with one dataset or metric, may not be optimal for another biological question, instrument type, acquisition method, or analysis goal.

Users should consider:

- The benchmark module and dataset.
- The software version.
- The submitted parameters (e.g. FDR)
- The relevant metric definitions.
- The trade-off between sensitivity and accuracy.
- Whether the workflow resembles their own intended use case.

ProteoBench is best used as a transparent decision-support resource rather than as a single definitive ranking system.
ProteoBench does not point to a single best one-fits-all data analysis workflow; should not be used as evidence for generalized statements about a workflow’s performance, and should not be used by developers as single performance measure of their workflow.

> ## How can I follow updates or community discussions?

Reviewer: ```suggestion
```

### For GitHub users

GitHub users can follow ProteoBench development through the main [ProteoBench GitHub repository](https://github.com/Proteobench/ProteoBench). This repository contains the source code, contribution information, issue tracking, and release history. Users who want to receive updates can watch or star the repository. If you are interested in a specific module, you can also watch the corresponding result repository listed [here](https://github.com/Proteobench).

Questions, proposals, and module-specific discussions can be followed through the [ProteoBench GitHub Discussions](https://github.com/Proteobench/ProteoBench/discussions) forum. This is the preferred route for users who want to participate in technical discussions, suggest new benchmark modules, discuss parser development, or contribute to the platform. Through this route, users can also register for discussion updates.

> ### For non-GitHub users

Reviewer: maybe we can add a link to the linkedin? And suggest that people can always contact us by email?
